app/controllers/api/v2/discovery_rules_controller.rb

```diff
@@ -31,7 +31,7 @@ def show
           param :name, String, :required => true
           param :search, String, :required => true
           param :hostgroup_id, Integer, :required => true
-          param :hostname, String, :required => true
+          param :hostname, String
```

Comment:
@lzap is it still required? I think we changed it once we allowed blank:
https://github.com/theforeman/foreman_discovery/blob/develop/app/models/discovery_rule.rb#L12